src/internal/connector/support/m365Transform.go

```diff
@@ -119,3 +119,95 @@ func insertStringToBody(body getContenter, newContent string) string {
 
 	return newContent + content
 }
+
+// ToListable utility function to encapsulate stored data for restoration.
+// New Listable omits trackable fields such as `id` or `ETag` and other read-only
+// objects that are prevented upon upload. Additionally, read-Only columns are
+// not attached in this method.
+// ListItems not included in creation of new list. ListItems  have to be restored
+// in separate call.
+func ToListable(orig models.Listable, displayName string) models.Listable {
+	newList := models.NewList()
+
+	newList.SetContentTypes(orig.GetContentTypes())
+	newList.SetCreatedBy(orig.GetCreatedBy())
+	newList.SetCreatedByUser(orig.GetCreatedByUser())
+	newList.SetCreatedDateTime(orig.GetCreatedDateTime())
+	newList.SetDescription(orig.GetDescription())
+	newList.SetDisplayName(&displayName)
+	newList.SetLastModifiedBy(orig.GetLastModifiedBy())
+	newList.SetLastModifiedByUser(orig.GetLastModifiedByUser())
+	newList.SetLastModifiedDateTime(orig.GetLastModifiedDateTime())
+	newList.SetList(orig.GetList())
+	newList.SetOdataType(orig.GetOdataType())
+	newList.SetParentReference(orig.GetParentReference())
+
+	columns := make([]models.ColumnDefinitionable, 0)
+	leg := make(map[string]struct{})
+	empty := struct{}{}
+	legacyNames := []string{"Attachments", "Edit", "Content Type"}
+
+	for _, entry := range legacyNames {
+		leg[entry] = empty
+	}
```

Comment:
This is a roundabout way of plainly creating a map.
```suggestion
	leg := map[string]struct{}{
		"Attachments": {},
		"Edit": {},
		"Content Type": {},
	}
```

Reply:
Resolved within [029cc9d](https://github.com/alcionai/corso/pull/2012/commits/029cc9d6a9256c1eb333294c93c0e1f35565e58d)